packages/hash/frontend/src/shared/ui/select.tsx

```diff
@@ -6,71 +6,78 @@ import {
   FormHelperText,
   InputLabel,
   outlinedInputClasses,
+  // eslint-disable-next-line no-restricted-imports
   Select as MuiSelect,
   SelectProps as MuiSelectProps,
   Typography,
 } from "@mui/material";
-import { FC, forwardRef, ReactNode } from "react";
+import { forwardRef, ReactNode, Ref, ReactElement } from "react";
 import { FontAwesomeIcon } from "../icons";
 
-export type SelectProps = {
+export type SelectProps<T = unknown> = {
   children: ReactNode;
   error?: boolean;
   helperText?: ReactNode;
   sx?: BoxProps["sx"];
   selectSx?: MuiSelectProps["sx"];
-} & Omit<MuiSelectProps, "sx">;
+} & Omit<MuiSelectProps<T>, "sx">;
 
-// @todo add custom class for component so it's easier to target the
-// component using css
+/**
+ * @todo add custom class for component so it's easier to target the component using css
+ */
 
-// For some reasons, renderValue prop doesn't properly infer types
-// when we use this wrapper Select component. But works properly
-// when the default Select component from MUI is used
-// see select-with-search-checkbox.tsx in playground.page
-// @todo figure out why this happens and fix
-
-export const Select: FC<SelectProps> = forwardRef(
-  (
-    { children, sx = [], selectSx = [], error, helperText, label, ...props },
-    ref,
-  ) => {
-    return (
-      <Box sx={sx}>
-        {label && (
-          <InputLabel {...(props.labelId && { id: props.labelId })}>
-            {label}
-          </InputLabel>
-        )}
-        <MuiSelect
-          fullWidth
-          sx={[
-            ({ palette }) => ({
-              ...(error && {
-                [`.${outlinedInputClasses.notchedOutline}`]: {
-                  borderColor: `${palette.orange[50]} !important`,
-                },
-              }),
+const Select = <T,>(
```

Comment:
```suggestion
const Select = <T>(
```

?

Reply:
Without that, ts thinks it's a JSX element without a closing tag. 

<img width="479" alt="Screenshot 2022-05-13 at 13 57 31" src="https://user-images.githubusercontent.com/16928688/168288151-3ab0d551-5571-4649-8d17-49b3320762bc.png">


We do something similar in the Suggester component => https://github.com/hashintel/hash/blob/a0ebc5f54bee9b9e8a1649ad76fd2a30d99deb66/packages/hash/frontend/src/blocks/page/createSuggester/Suggester.tsx#L20